src/jacobians.jl

```diff
@@ -120,7 +120,7 @@ end
 
 
 function _make_Ji(::AbstractArray, xtype, dx, color_i, nrows, ncols)
-    Ji = mapreduce(i -> i==color_i ? dx : convert(typeof(dx), zeros(xtype,nrows)), hcat, 1:ncols)
+    Ji = mapreduce(i -> i==color_i ? dx : (ret = similar(dx, nrows); fill!(ret, 0.)), hcat, 1:ncols)
```

Comment:
this will change the type if it's a static array.

Reply:
If it's a static array, and you're creating a regular array and adding it to a static array, you will get a regular array

Reply:
We should probably add a special case for static arrays, if this function can be called with a static array.